internal/storage/rocksdb/store.go

```diff
@@ -515,7 +592,17 @@ func (rdb *rocksDB) getSingleKey(ro *gorocksdb.ReadOptions, key []byte) ([]*serv
 	defer value.Free()
 	val := toByteArray(value)
 	if val != nil && len(val) > 0 {
-		return []*serverpb.KVPair{&serverpb.KVPair{Key: key, Value: val}}, nil
+		//check ttl.
+		expireTs, _ := getExpireTs(val)
+		if expireTs > 0 && expireTs < time.Now().Unix() {
+			//trigger delete.
+			rdb.Delete(key)
```

Comment:
Do we need to trigger a delete here ? So this effectively converts a read-only call to a read-write one. Can we just check the expiry and not return (nil) if expired ?

Reply:
Removed this.